### ISSUE TITLE
Microsoft Edge dev does not uninstall microsoft autoupdate

### ISSUE THREAD

**Issue by sashkab** [Homebrew/homebrew-cask-versions#7630]
#### General troubleshooting steps

- [x] I understand that [if I ignore these instructions, my issue may be closed without review](https://github.com/Homebrew/homebrew-cask/blob/master/doc/faq/closing_issues_without_review.md).
- [x] I have retried my command with `--force`.
- [x] I ran `brew update-reset && brew update` and retried my command.
- [x] I ran `brew doctor`, fixed as many issues as possible and retried my command.
- [x] I have checked the instructions for [reporting bugs](https://github.com/Homebrew/homebrew-cask#reporting-bugs).
- [x] I made doubly sure this is not a [checksum does not match](https://github.com/Homebrew/homebrew-cask/blob/master/doc/reporting_bugs/checksum_does_not_match_error.md) error.

#### Description of issue

When installing Microsoft Edge dev, it installs auto-update daemon. After running `brew cask remove microsoft-edge-dev`, auto-update daemon stays on and continues to run.

#### Command that failed

Command did not fail, but failed to cleanup system properly.

#### Proper way to uninstall autoupdate agent

```sh
launchctl stop com.microsoft.update.agent
launchctl remove com.microsoft.update.agent
sudo launchctl stop com.microsoft.update.agent
sudo launchctl remove com.microsoft.update.agent
sudo launchctl stop com.microsoft.autoupdate.helper
sudo launchctl remove com.microsoft.autoupdate.helper
sudo rm /Library/PrivilegedHelperTools/com.microsoft.autoupdate.helper
sudo rm -r /Library/Application\ Support/Microsoft
```

#### Things to consider

If Microsoft Office is installed, it might be a good idea to leave autoupdate on. Otherwise, if Edge was final product -- remove it.

#### brew doctor

```sh
$ brew doctor
Your system is ready to brew.
```

**Comment by reitermarkus** [comment]
It can be added to `zap` instead of `uninstall`, as in `microsoft-office`.

**Comment by coreygo** [comment]
[Microsoft Remote Desktop Beta](https://github.com/Homebrew/homebrew-cask-versions/blob/master/Casks/microsoft-remote-desktop-beta.rb) also uses the update helper, as does [Microsoft Edge Canary](https://github.com/Homebrew/homebrew-cask-versions/blob/master/Casks/microsoft-edge-canary.rb).

Is there a recommended mechanism to detect whether the Microsoft Update Agent is still being used by other Microsoft apps? As mentioned under things to consider, it seems like it should be removed iff it's an orphaned dependency.

**Comment by vitorgalvao** [comment]
> As mentioned under things to consider, it seems like it should be removed iff it's an orphaned dependency.

That would make the cask more complex than it’s worth, and it’s another thing to fail. As @reitermarkus mentioned, add it to `zap` — that’s the recommended standard method.

**Comment by coreygo** [comment]
> more complex than it’s worth, and it’s another thing to fail

Indeed. Thanks for confirming the recommended method.

**Comment by coreygo** [comment]
Resolved with #7777.
